File: aegis/sdk/go/aegis/conformance.go
package aegis

import "strings"

func ValidateFixture(path string) (ValidationResult, error) {
	if strings.HasSuffix(path, ".packet.json") {
		packet, err := LoadPacket(path)
		if err != nil {
			return ValidationResult{}, err
		}
		return ValidatePacket(packet), nil
	}
	if strings.HasSuffix(path, ".provenance.json") {
		envelope, err := LoadProvenanceEnvelope(path)
		if err != nil {
			return ValidationResult{}, err
		}
		return ValidateProvenanceEnvelope(envelope), nil
	}
	manifest, err := LoadManifest(path)
	if err != nil {
		return ValidationResult{}, err
	}
	return ValidateManifest(manifest), nil
}

File: aegis/sdk/python/aegis/conformance.py
from pathlib import Path

from .manifest import load_manifest, validate_manifest
from .packet import load_packet, validate_packet
from .provenance import load_provenance_envelope, validate_provenance_envelope
from .validation import ValidationResult


def validate_fixture(path: str | Path) -> ValidationResult:
    fixture_path = Path(path)
    if fixture_path.name.endswith(".packet.json"):
        return validate_packet(load_packet(fixture_path))
    if fixture_path.name.endswith(".provenance.json"):
        return validate_provenance_envelope(load_provenance_envelope(fixture_path))
    return validate_manifest(load_manifest(fixture_path))
